FinalTask/src/main/java/driver/Driver.java

```diff
@@ -0,0 +1,108 @@
+package driver;
+
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.net.URL;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+public class Driver {
+    private static final String DESTINATION_URL = "https://gmail.com";
+    private static final String USERNAME = "olgastarkova";
+    private static final String ACCESS_KEY = "06c917f0-296f-4837-9e77-aac8a4289310";
+    public static final String CLOUD_URL = "https://" + USERNAME + ":" + ACCESS_KEY + "@ondemand.saucelabs.com:443/wd/hub";
+    private DesiredCapabilities desiredCaps;
+    private Capabilities cap;
+    private static Driver instance = null;
+    private WebDriver driver;
+    private DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+    private Date date;
+    private static final int RUN_METHOD = 2; //1 = locally, 2 = cloud, 3 = selenium grid
+
+
+    private Driver() {}
+
+    public static Driver getInstance() {
+        if (instance == null) {
+            synchronized (Driver.class) {
+                if (instance == null) {
+                    instance = new Driver();
+                }
+            }
+        }
+        return instance;
+    }
+
+    public WebDriver getDriver() {
+        return driver;
+    }
+
+    public WebDriver open(String browser) throws Exception {
```

Comment:
Method is too long, you can create 3 private methods and this method will be more readable